### ISSUE TITLE
Charts - ERROR: Attempting to index local 'self'; charts.lua:256

### ISSUE THREAD

**Issue by TheMagikPeanut** [OpenPrograms/Fingercomp-Programs#17]
Any time I construct a container and call the draw function, I get this error about attempting to index local self, pointing to line 256 of charts.lua. Upon inspection of charts.lua, that whole code block is filled with references to self. I'm not a skilled coder (yet), but as far as I can tell, I can't see anything wrong with the self referencing of the object. I looks just like anything I see in other (non-oc) lua code in self-referencing by an object of itself in lua OOP.

I'm using (or attempting to) whatever version of charts oppm will install (I couldn't get hpm to install it due to an HTTP response error; is the repo down?). Open Computers 1.7.5.192 on Minecraft 1.12.2.

 This is what I get.
|  /usr/lib/charts.lua:256: attempt to index local 'self' (a nil value):
|  stack traceback:
|  /usr/lib/charts.lua:256: in fuction 'draw'
|  /usr/bin/battpower.lua:105: in main chunk <-- this is obviously my program
|  (...tail calls...)
|  [C]: in function 'xpcall'
|  machine:798: in function 'xpcall'
|  /lib/process.lua:63: in functtion </lib/process.lua:59>

EDIT: Fixed error not on new line

P.S.: Didn't attach as a screenshot, because the game is running, and using a different pc for this.

**Comment by TheMagikPeanut** [comment]
Thank you so much! I've learned more about that damn colon in the last few hours than in my entire time writing lua (only a few years, and OC is the only reason I started)... I don't know why I didn't think of that after learning about that colon, and how it works. Kinda makes sense, now that I think about it, that its instance would need to be passed back to itself to function that way as an object, hence the lack of classes in Lua. Or at least that's my uneducated understanding.

Thank you for your speedy reply. Will you be adding to this library in the future?
